src/main/java/com/google/sps/data/Room.java

```diff
@@ -18,16 +19,80 @@
 import java.util.Map;
 import java.util.stream.*;
 
+//Object repsresenting the Room that the chat and video streaming will be in
+//TODO: Restgructure so logic is more split up
 public class Room {
     private static final String ROOM_ENTITY = "Room";
     private static final String MEMBERS_PROPERTY = "members";
     private static final String MESSAGES_PROPERTY = "messages";
     private static final String VIDEOS_PROPERTY = "videos";
+    public static final int MAX_MESSAGES = 10;
     private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     private List<Member> members;
-    private LinkedList<Message> messages;
+    private Queue<Message> messages;
     private Queue<Video> videos;
 
+    //Room factory function
+    public static Room createRoom(List<Member> members, Queue<Video> videos, Queue<Message> messages){
+        return new Room(members, videos, messages);
+
+    }
+
+    // Manipulates the messages property of the room given a room key and a new message to be added
+    public static void addMessagesFromKey(Key roomKey, Message chatMessage) {
```

Comment:
Other files shouldn't need to make a roomKey. Just have them pass a roomID.

Also, this logic should be done inside the Servlet and not in Room. (adding messages)

The datastore put should be done in this file. Here is logic:

Servlet gets roomID -> get Room Obj from Room.class -> add message -> call Room.storeRoom(Room) to save in datastore.

Reply:
I didn't create this function. This was created by Dangely for her feature.